GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/ColorShader.java

```diff
@@ -0,0 +1,39 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.gearvrf.particlesystem;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRShaderTemplate;
+import org.gearvrf.utility.TextFile;
+
+public class ColorShader extends GVRShaderTemplate
```

Comment:
Doing this gives the following error:
java.lang.UnsupportedOperationException: error creating shader template of class ParticleShaderjava.lang.Class<org.gearvrf.particlesystem.ParticleShader> is not accessible from java.lang.Class<org.gearvrf.GVRBaseShaderManager>
The call causing this error is "retrieveShaderTemplate" in from an instance of GVRMaterialShaderManager in Particles.java.
`        GVRMaterialShaderManager shaderManager = mGVRContext.getMaterialShaderManager();
        GVRShaderTemplate shaderTemplate = shaderManager.retrieveShaderTemplate(ParticleShader.class);
        shaderTemplate.bindShader(mGVRContext, meshObject.getRenderData(), mGVRContext.getMainScene());`
Looks like any custom shader class needs to be visible to the framework.